### PR TITLE
Pinot bugs and making it work with the latest trunk pinot

### PR DESCRIPTION
**Pull request by agrawaldevesh**
Three changes here: 1 medium and 2 tiny:
* The nameToIndex used for the pinot column ordering map should not assume that the variable reference name is the same as the pinot lowercase name. It should be keyed off the keys in the selections map (in the pinot query generator context) and not the value. It was throwing an error with the query below because the output variables were named like player_initial_33, g_old_51 etc instead of just player_initial, g_old. This is because these columns are referenced twice in the query. 
This again smells like a core bug and it is totally my bad that I screwed this up the last time. 

```
select * from (select yearid, substr(playerid, 1, 3) as player_initial, sum(homeruns) from pinot.baseballStats.baseballStats group by 1, 2) as m INNER join (select g_old, substr(playername, 1, 3) as player_name_initial, sum(numberofgames) from pinot.baseballStats.baseballStats group by 1, 2) as w ON m.player_initial = w.player_name_initial;
```

* [tiny] Make this work with the new pinot trunk code which no longer likes "Content-Type: Application/json" for just GET requests and non body POSTS :-). As it shouldn't. 
* [tiny] renamed the config prefer broker queries to ! forbid broker queries to be true to the name

```
== RELEASE NOTES ==

Pinot Changes
*  Replace config `pinot.prefer-broker-queries` with the inverse config `pinot.forbid-broker-queries`.

### CODE REVIEW

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java [hunk 1]
```diff
@@ -338,11 +338,11 @@ else if (!groupByColumns.isEmpty()) {
...
         }
 
-        List<PinotColumnHandle> columnHandles = ImmutableList.copyOf(getAssignments().values());
-        return new PinotQueryGenerator.GeneratedPql(tableName, query, getIndicesMappingFromPinotSchemaToPrestoSchema(query, columnHandles), groupByColumns.size(), filter.isPresent(), isQueryShort);
+        List<Integer> indices = getIndicesMappingFromPinotSchemaToPrestoSchema(query, getAssignments());
+        return new PinotQueryGenerator.GeneratedPql(tableName, query, indices, groupByColumns.size(), filter.isPresent(), isQueryShort);
     }
 
-    private List<Integer> getIndicesMappingFromPinotSchemaToPrestoSchema(String query, List<PinotColumnHandle> handles)
+    private List<Integer> getIndicesMappingFromPinotSchemaToPrestoSchema(String query, LinkedHashMap<VariableReferenceExpression, PinotColumnHandle> assignments)
```
Comment (highker) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java]: nit: `LinkedHashMap` -> `Map`

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGenerator.java [hunk 2]
```diff
@@ -130,10 +130,10 @@ public PinotQueryGeneratorContext getContext()
     public Optional<PinotQueryGeneratorResult> generate(PlanNode plan, ConnectorSession session)
     {
         try {
-            boolean preferBrokerQueries = PinotSessionProperties.isPreferBrokerQueries(session);
-            PinotQueryGeneratorContext context = requireNonNull(plan.accept(new PinotQueryPlanVisitor(session, preferBrokerQueries), new PinotQueryGeneratorContext()), "Resulting context is null");
-            boolean isQueryShort = context.isQueryShort(PinotSessionProperties.getNonAggregateLimitForBrokerQueries(session));
-            return Optional.of(new PinotQueryGeneratorResult(context.toQuery(pinotConfig, preferBrokerQueries, isQueryShort), context));
+            PinotQueryGeneratorContext context = requireNonNull(plan.accept(new PinotQueryPlanVisitor(session),
```
Comment (highker) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGenerator.java]: move `plan.accept(new PinotQueryPlanVisitor(session)` to its own line

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java [hunk 3]
```diff
@@ -275,7 +277,9 @@ private boolean hasOrderBy()
                 .filter(s -> !groupByColumns.contains(s.getKey())) // remove the group by columns from the query as Pinot barfs if the group by column is an expression
                 .map(s -> s.getValue().getDefinition())
                 .collect(Collectors.joining(", "));
-
+        if (expressions.isEmpty()) {
+            throw new PinotException(PINOT_QUERY_GENERATOR_FAILURE, Optional.empty(), "No valid expressions in " + this);
+        }
```
Comment (highker) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java]: break an empty line after this

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java [hunk 4]
```diff
@@ -275,7 +277,9 @@ private boolean hasOrderBy()
                 .filter(s -> !groupByColumns.contains(s.getKey())) // remove the group by columns from the query as Pinot barfs if the group by column is an expression
                 .map(s -> s.getValue().getDefinition())
                 .collect(Collectors.joining(", "));
-
+        if (expressions.isEmpty()) {
+            throw new PinotException(PINOT_QUERY_GENERATOR_FAILURE, Optional.empty(), "No valid expressions in " + this);
```
Comment (highker) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java]: `this` will provide the memory address (which is not necessary for an error message). What is needed is `PinotQueryGeneratorContext.class.getName()` I guess.

**Comment by agrawaldevesh** [comment]
James, thanks for the comments and careful review. As always !.

**Pull request merged**